packages/cli-kit/src/public/node/hooks/prerun.ts

```diff
@@ -81,3 +86,18 @@ function findAlias(aliases: string[]) {
     return existingAlias.replace(/:/g, ' ')
   }
 }
+
+function warnOnAvailableUpgrade() {
+  const cliDependency = '@shopify/cli'
+  const currentVersion = CLI_KIT_VERSION
+
+  // Check in the background, once daily
+  void checkForNewVersion(cliDependency, currentVersion, {cacheExpiryInHours: 24})
+
+  // Warn if we previously found a new version
+  const lastVersion = checkForCachedNewVersion(cliDependency, currentVersion)
```

Comment:
Maybe I'm missing something, but can't we get the new version already from the previous line? Do we really need that new method to only retrieve (without repopulate)?

```suggestion
  const lastVersion = checkForNewVersion(cliDependency, currentVersion, {cacheExpiryInHours: 24})
```

Reply:
TIL! What happens if the command finishes before that unawaited promise resolves? Will it wait for it? Have you tested it?

Reply:
It will await the unresolved promise before exiting. This is just a single request to npmjs, and of course ctrl+c should kill it, so in nearly all cases this won't be noticeable.